common/djangoapps/microsite_configuration/backends/filebased.py

```diff
@@ -0,0 +1,199 @@
+        This will return if current request is a request within a microsite
+        """
+        return bool(self.get_configuration())
+
+    def has_override_value(self, val_name):
+        """
+        Will return True/False whether a Microsite has a definition for the
+        specified val_name
+        """
+        configuration = self.get_configuration()
+        return val_name in configuration
+
+    def get_all_config(self):
+        """
+        This returns all configuration for all microsites
+        """
+        config = {}
+
+        for key, value in settings.MICROSITE_CONFIGURATION.iteritems():
+            config[key] = value
+
+        return config
+
+    def get_value_for_org(self, org, val_name, default=None):
+        """
+        This returns a configuration value for a microsite which has an org_filter that matches
+        what is passed in
+        """
+
+        if not self.has_configuration_set():
+            return default
+
+        # Filter at the setting file
+        for value in settings.MICROSITE_CONFIGURATION.itervalues():
+            org_filter = value.get('course_org_filter', None)
+            if org_filter == org:
+                return value.get(val_name, default)
+        return default
+
+    def get_all_orgs(self):
+        """
+        This returns a set of orgs that are considered within a microsite. This can be used,
+        for example, to do filtering
+        """
+        org_filter_set = set()
+
+        # cdodge: is this really needed? Right now it could be expensive as it does
+        # a full table count...
+        if not self.has_configuration_set():
+            return org_filter_set
+
+        # Get the orgs in the db
+        for microsite in settings.MICROSITE_CONFIGURATION.itervalues():
+            org_filter = microsite.get('course_org_filter')
+            if org_filter:
+                org_filter_set.add(org_filter)
+
+        return org_filter_set
+
+    def _set_microsite_config(self, microsite_config_key, subdomain, domain):
+        """
+        Helper internal method to actually find the microsite configuration
+        """
+        config = settings.MICROSITE_CONFIGURATION[microsite_config_key].copy()
+        config['subdomain'] = subdomain
+        config['microsite_config_key'] = microsite_config_key
+        config['site_domain'] = domain
+
+        template_dir = settings.MICROSITE_ROOT_DIR / microsite_config_key / 'templates'
+        config['template_dir'] = template_dir
+        self.current_request_configuration.data = config
+
+    def clear(self):
+        """
+        Clears out any microsite configuration from the current request/thread
+        """
+        self.current_request_configuration.data = {}
+        self.current_request_configuration.cache = {}
+
+    def enable_microsites(self, log):
+        """
+        Enable the use of microsites, from a dynamic defined list in the db
+        """
+        if not settings.FEATURES['USE_MICROSITES']:
```

Comment:
This line makes me think we should add an is_feature_enabled() operation and reference it where applicable throughout the changeset